### PR TITLE
document user endpoints

### PR DESCRIPTION
**Pull request by oblakeerickson**
- document user endpoints, allow for empty request/response bodies
- document more user endpoints, improve debugging output if no details are specified
- document some more user endpoints

### CODE REVIEW

File: spec/requests/api/shared/shared_examples.rb [hunk 1]
```diff
@@ -21,21 +27,25 @@ def expect_schema_valid(schemer, params)
...
-      schemer = JSONSchemer.schema(expected_request_schema)
-      expect_schema_valid(schemer, params)
+    describe "request body" do
+      it "matches the documented request schema" do |example|
+        if expected_request_schema
+          schemer = JSONSchemer.schema(expected_request_schema)
+          expect_schema_valid(schemer, params)
+        end
+      end
     end
```
Comment (CvX) [inline spec/requests/api/shared/shared_examples.rb:37]: 🧑‍🎨 (indentation)

```suggestion
  describe "request body" do
    it "matches the documented request schema" do |example|
      if expected_request_schema
        schemer = JSONSchemer.schema(expected_request_schema)
        expect_schema_valid(schemer, params)
      end
    end
  end
```

File: spec/requests/api/shared/shared_examples.rb [hunk 2]
```diff
@@ -21,21 +27,25 @@ def expect_schema_valid(schemer, params)
...
 
     it "matches the documented response schema" do  |example|
-      schemer = JSONSchemer.schema(
-        expected_response_schema,
-      )
-      expect_schema_valid(schemer, json_response)
+      if expected_response_schema
+        schemer = JSONSchemer.schema(
+          expected_response_schema,
+        )
```
Comment (CvX) [inline spec/requests/api/shared/shared_examples.rb:46]:
```suggestion
        schemer = JSONSchemer.schema(expected_response_schema)
```

File: spec/requests/api/shared/shared_examples.rb [hunk 3]
```diff
@@ -21,21 +27,25 @@ def expect_schema_valid(schemer, params)
...
+          expect_schema_valid(schemer, params)
+        end
+      end
     end
-  end
 
   describe "response body" do
     let(:json_response) { JSON.parse(response.body) }
 
     it "matches the documented response schema" do  |example|
```
Comment (CvX) [inline spec/requests/api/shared/shared_examples.rb:42]:
```suggestion
    it "matches the documented response schema" do |example|
```

**Review by CvX — APPROVED** [review]
👍, just a couple of v. minor code style suggestions

**Comment by oblakeerickson** [comment]
@CvX thank you!

**Pull request merged**